packages/syft/src/syft/service/worker/worker_service.py

```diff
@@ -160,17 +160,19 @@ def list(self, context: AuthedServiceContext) -> Union[SyftSuccess, SyftError]:
             return workers
 
         # If container workers, check their statuses
-        for idx, worker in enumerate(workers):
-            result = check_and_update_status_for_worker(
-                worker=worker,
-                worker_stash=self.stash,
-                credentials=context.credentials,
-            )
-            if result.is_err():
-                return SyftError(
-                    message=f"Failed to update status for worker: {worker.id}. Error: {result.err()}"
+        with contextlib.closing(docker.from_env()) as client:
+            for idx, worker in enumerate(workers):
+                worker_ = _check_and_update_status_for_worker(
+                    client=client,
+                    worker=worker,
+                    worker_stash=self.stash,
+                    credentials=context.credentials,
                 )
-            workers[idx] = worker
```

Comment:
@shubham3121 There was a bug here. Earlier this was
```py
for idx, worker in enumerate(workers):
    result = check_and_update_status_for_worker(
    ...
workers[idx] = worker
```

`workers[idx]` was not updated correctly, still using the one with outdated status.

Reply:
Good catch 👍🏾 . Thanks for the fix.